### PR TITLE
feat: face opponents base on spawn

### PR DESCRIPTION
**Pull request by ahv15**
Requires https://github.com/MovingBlocks/Terasology/pull/4823 to test this.

### CODE REVIEW

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/PlayerDeathSystem.java [hunk 1]
```diff
@@ -65,6 +66,8 @@ public void beforeDestroy(BeforeDestroyEvent event, EntityRef player, CharacterC
             player.send(new RestoreFullHealthEvent(player));
             Vector3f randomVector = new Vector3f(-1 + random.nextInt(3), 0, -1 + random.nextInt(3));
             player.send(new CharacterTeleportEvent(randomVector.add(LASUtils.getTeleportDestination(team))));
+            float playerYaw = team.equals(LASUtils.RED_TEAM) ? -90 : 90;
```
Comment (skaldarnar) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/PlayerDeathSystem.java]: While this works currently for our simple map, this code would be more resilient if we'd compute the yaw from the teleport target position and the opponent player base position.

**Review by skaldarnar — COMMENTED** [review]
The fact that we have to change code in three places worries me a bit. We should probably take a look at how we can refactor this.

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/PlayerDeathSystem.java (hunk 1 above)
Comment (ahv15) [reply src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/PlayerDeathSystem.java]: Made use of this https://stackoverflow.com/questions/5188561/signed-angle-between-two-3d-vectors-with-same-origin-within-the-same-plane as the default calculation of angles isn't signed. The 2nd comment gave me a more precise answer so I used that.

File: src/main/java/org/terasology/ligthandshadow/componentsystem/LASUtils.java [hunk 2]
```diff
@@ -178,6 +178,14 @@ public static String getOppositionTeam(String team) {
         return null;
     }
 
+    public static float getYaw(String team) {
```
Comment (jdrueckert) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/LASUtils.java]: `getYaw` seems a good name for a general utility function, but doesn't match the hard-coded (and not documented) "assumptions" about _what_ yaw we're calculating here (relation to the teleport destinations of the teams, etc.)

**Review by skaldarnar — APPROVED** [review]

**Comment by skaldarnar** [comment]
Builds are failing because of https://github.com/Terasology/LightAndShadowResources/pull/70#issuecomment-883678945

**Pull request merged**